### PR TITLE
feat!(cpoptions): remove "n" flag

### PR DESCRIPTION
**Pull request by bfredl**
This was a stupid about of code for a "feature" which just makes the screen look ugly as fuck.

### CODE REVIEW

**Comment by luukvbaal** [comment]
I mean I don't think it's particularly pretty either but:
1. It's subjective; users of this option came out of the woodwork on the initial 'statuscolumn' PR as well for example.
2. Makes porting 'smoothscroll' (still quite unfinished) patches from Vim significantly harder on the off-chance that 'smoothscroll' gets improved further over at vim/vim without Bram. I don't think it's worth straying from Vim over this (is it really that much code?).

**Comment by bfredl** [comment]
> It's subjective; users of this option came out of the woodwork on the initial 'statuscolumn' PR as well for example.

Well, there is an objective component which is: this legacy feature can _only_ affect number while not supporting any of the newly added columns. it still _works_ with them but it only works for users who "subjectively" want to special case number column over all other leftcolumns.

>  Although one could argue we have 'breakindent[opt]' for that.

The issue here is that **this** particular feature uses a very different code path than all other such features. One could even prototype an "anti-breakindent" (extra indent on the first line) feature as a plugin, by using inline extmarks, although with likely perf/correctness issues compared to a builtin anti-breakindent (or a more flexible plugin mechanism, like full support for ephemeral decorations in plines.c and/or fully dynamic 'statuscol').

**Comment by bfredl** [comment]
> Makes porting 'smoothscroll' (still quite unfinished) patches from Vim significantly harder on the off-chance that 'smoothscroll' gets improved further over at vim/vim without Bram. I don't think it's worth diverging from Vim over this (is it really that much code?).

This is a direct contradiction in terms. If this change is "not that much code" when considered as a clean-up, it is also **not that much code** when it affects vim-patches.

Also the PR is about `cpo+=n`. Any patch which affects that, or its interaction with 'smoothscroll' or another feature, could just as easily become N/A instead.  It sounds quite byzantine to keep features around just so that we can port patches for bugs in that feature, instead of not just having those bugs in the first place..

**Comment by luukvbaal** [comment]
> This is a direct contradiction in terms. If this change is "not that much code" when considered as a clean-up, it is also not that much code when it affects vim-patches.

OK let me rephrase. Is it really that big of a cleanup that it is worth diverging from Vim over. Where it is tightly intertwined with 'smoothscroll' code which can be expected to receive further updates to make it work better.

> Also the PR is about cpo+=n. Any patch which affects that, or its interaction with 'smoothscroll' or another feature, could just as easily become N/A instead. It sounds quite byzantine to keep features around just so that we can port patches for bugs in that feature, instead of not just having those bugs in the first place..

I wasn't saying that such patches can't be moved if we were to drop support for cpo+=n, just that it makes it (unnecessarily) harder. I was just playing devil's advocate and arguing that it might not be worth it. I don't care for the option itself.

**Comment by bfredl** [comment]
I just don't see what would be the "hard" part (even as potential risk). Like, at all. Would deleting dead branches and folding "width2" into "width1" be dramatically harder in _new_ smoothscroll code comparing to what it was is existing code? It is of course possible, as in any yet unwritten patch is possible, but it sounds too far fetched and vague to cancel out the real immediate benefits of an actually performed change to make the code easier to read and work with in _neovim_ .

**Comment by luukvbaal** [comment]
Sure, I also don't think the difficulty in porting Vim patches is a particulary strong argument against deprecating features in general. Just thought it was worth mentioning in this case since 'smoothscroll' is quite new... TBH the first two arguments I listed are stronger. If we can provide an alternative (through plugins or otherwise), I'm all for it. The width2 stuff was indeed very confusing/hard to read while I was porting 'smoothscroll'.
